tests/test_query.py

```diff
@@ -619,10 +619,38 @@ def test_query_without_token_succeeds_when_auth_disabled(
     mock_successful_get,
     monkeypatch,
     disable_auth,
+    set_test_credentials,
 ):
     """
     Test that when authentication is disabled, a request to the /query route without a token succeeds.
     """
     monkeypatch.setattr(crud, "get", mock_successful_get)
     response = test_app.get(ROUTE)
     assert response.status_code == 200
+
+
+@pytest.mark.integration
+def test_integration_query_without_auth_succeeds(
+    test_app, monkeypatch, disable_auth, set_test_credentials
+):
+    """
+    Running a test against a real local test graph
+    should succeed when authentication is disabled.
+    """
+    monkeypatch.setattr(
+        util, "GRAPH_DB", util.EnvVar(util.GRAPH_DB.name, "repositories/my_db")
+    )
```

Comment:
Yes, good point. I think when I wrote the test I was focused on getting it to pass and didn't think we'd actually commit it to the test suite (which we now are). I honestly don't like this way of mocking the app state and given how verbose it is (and how much implementation details you have to be aware of to get the test to pass) I think it could be worth it to find a smarter way in a separate issue. 

I'll make an issue and add a comment.